app/models/teachable_parser.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+# Teachable parser class
+# This is a service PORO model that is used in the media search
+class TeachableParser
+  # params is a hash with key :teachable_ids
+  # teachable ids is an array made up of strings composed of 'Lecture-'
+  # or 'Course-' followed by the id
+  def initialize(params)
+    @teachable_ids = params[:teachable_ids] || []
+    @all_teachables = params[:all_teachables] == '1'
+    @inheritance = params[:teachable_inheritance] == '1'
+  end
+
+  # returns all courses, lectures and lessons that are associated
+  # to the given list of teachables
+  # depending on the teachable_inheritance parameter, this is done with or
+  # without inheritance (without inheritance just meaning that the input
+  # array of teachable_ids is returned)
+  # if the all_teachable flag is set to '1', it returns []
+  # results are returned in the form of strings:
+  # e.g. as ['Course-5', 'Lecture-2', 'Lesson-39']
+  def teachables_as_strings
+    return [] if @all_teachables
+    return @teachable_ids unless @inheritance
+
+    teachables_with_inheritance.map { |t| "#{t.class}-#{t.id}" }
+  end
+
+  private
+
+    def lecture_ids
+      @teachable_ids.select { |t| t.start_with?('Lecture') }
+                    .map { |t| t.remove('Lecture-') }
+    end
+
+    def lectures
+      Lecture.where(id: lecture_ids)
+    end
+
+    def course_ids
+      @teachable_ids.select { |t| t.start_with?('Course') }
+                    .map { |t| t.remove('Course-') }
```

Comment:
I don't know how type-safe this operation is, I would expect it to return an array of int and not string.

Reply:
Yeah, conceptionally it should definitely be an array of integers (even though ActiveRecord does not care and accepts strings for ids as well). I will add a `.map(&:to_i)`.